src/application/parser/NumberSplitter.js

```diff
@@ -0,0 +1,22 @@
+export default class NumberSplitter {
+  static split(delimiters, str) {
+    const result = str.split('').reduce((acc, char) => {
+      if (delimiters.includes(char)) {
+        if (acc.currentNumber) {
+          acc.numbers.push(parseInt(acc.currentNumber, 10));
+          acc.currentNumber = '';
+        }
+      } else {
+        acc.currentNumber += char;
+      }
+
+      return acc;
+    }, { numbers: [], currentNumber: '' });
```

Comment:
```suggestion
    const result = str.split('').reduce((acc, char) => {
      if (delimiters.includes(char) && acc.currentNumber) {
        acc.numbers = [...acc.numbers, paseInt(acc.currentNumber, 10));
        acc.currentNumber = '';
        return acc;
     }
     acc.currentNumber += char;
     
     return acc;
    }, { numbers: [], currentNumber: '' });
```

이렇게 작성하면 deps를 줄일 수 있지않을까요?

Reply:
음.. 이런경우에는 연속적으로 구분자가 나오는 경우에는 처리가 불가능해서 어쩔 수 없이 depth를 늘린 거긴 한데, depth를 줄일 수 있는 방안에 대해 더 고민해 보겠습니다!